src/utils/child_process.rs

```diff
@@ -89,8 +91,135 @@ impl Nanny {
 }
 
 fn boot_desktop_file(path: &Path) -> std::io::Result<Child> {
-    let args = format!( "`if [ \"$(grep '^X-GNOME-Autostart-enabled' {:?} | tail -1 | sed 's/^X-GNOME-Autostart-enabled=//' | tr '[A-Z]' '[a-z]')\" != 'false' ]; then grep '^Exec' {:?} | tail -1 | sed 's/^Exec=//' | sed 's/%.//' | sed 's/^\"//g' | sed 's/\" *$//g'; else echo 'exit'; fi`", path , path);
-    Command::new("sh").arg("-c").arg(args).spawn()
+    use std::io::Error as StdError;
+    let entry = DesktopEntry::parse_file(path)?;
+    let dw_env = std::env::var("XDG_CURRENT_DESKTOP").unwrap_or(Default::default());
+
+    if entry.only_show_in.is_some() {
+        if !entry.only_show_in.unwrap().contains(&dw_env) {
+            return Err(StdError::from(std::io::ErrorKind::Other));
+        }
+    }
+    if entry.not_show_in.is_some() {
+        if entry.not_show_in.unwrap().contains(&dw_env) {
+            return Err(StdError::from(std::io::ErrorKind::Other));
+        }
+    }
```

Comment:
This could also be a:
```rust
if entry.only_show_in.is_some() && !entry.only_show_in.unwrap().contains(&dw_env) {
     return Err(StdError::from(std::io::ErrorKind::Other));
}
if entry.not_show_in.is_some() && entry.not_show_in.unwrap().contains(&dw_env) {
     return Err(StdError::from(std::io::ErrorKind::Other));
}
```
right?

Reply:
Well it would be better to use 
```
    if let Some(only_show_in) = entry.only_show_in {
        if !only_show_in.contains(&dw_env) {
            return Err(StdError::from(std::io::ErrorKind::Other));
        }
    }
```
Then the same for the other if

Reply:
Ok, so what is the benefit of the cascaded `if`?

Reply:
I think it might error if the unwrap is in the same line as is_some when there is a None. As they would get checked at the same time.
Then it is still unstable to chain checks with if let Some.
Thanks.

Reply:
Thanks, for the explanation. I thought I once had clippy complaining about this structure, but if the conditions are in that kind of relation it makes sense to avoid this. 
TIL